src/core/__tests__/props-updating.test.tsx

```diff
@@ -208,112 +246,83 @@ describe("controlled mode", () => {
+            </ControlledComponent>
+        );
 
-        it("ignores nested option with default prefix", () => {
-            mount(
-                <ControlledComponent>
-                    <NestedComponent defaultC="default" />
-                </ControlledComponent>
-            );
+        expect(WidgetClass.mock.calls[0][1].nestedOption.c).toBe("default");
+        expect(WidgetClass.mock.calls[0][1].nestedOption).not.toHaveProperty("defaultC");
+    });
 
-            fireOptionChange("nestedOption.c", "changed");
-            jest.runAllTimers();
-            expect(Widget.option.mock.calls.length).toBe(0);
-        });
+    it("does not pass default values to widget if controlledOption set", () => {
+        shallow(
+            <ControlledComponent defaultControlledOption={"default"} controlledOption={"controlled"} />
+        );
 
-        it("ignores 3rd-party changes in default props", () => {
-            const component = shallow(
-                <ControlledComponent defaultControlledOption={"default"} />
-            );
-            component.setProps({
-                defaultControlledOption: "changed"
-            });
-            jest.runAllTimers();
-            expect(Widget.option.mock.calls.length).toBe(0);
-        });
+        expect(Widget.option.mock.calls.length).toBe(0);
+        expect(WidgetClass.mock.calls[0][1].controlledOption).toBe("controlled");
+        expect(WidgetClass.mock.calls[0][1]).not.toHaveProperty("defaultControlledOption");
+    });
 
-        it("ignores 3rd-party changes in nested default props", () => {
-            const component = shallow(
-                <ControlledComponent>
-                    <NestedComponent defaultC="default" />
-                </ControlledComponent>
+    it("ignores nested option with default prefix", () => {
+        mount(
+            <ControlledComponent>
+                <NestedComponent defaultC="default" />
+            </ControlledComponent>
+        );
 
-            );
-            const nested = component.find(NestedComponent).dive();
+        fireOptionChange("nestedOption.c", "changed");
+        jest.runAllTimers();
+        expect(Widget.option.mock.calls.length).toBe(0);
+    });
 
-            nested.setProps({
-                defaultC: "changed"
-            });
-            jest.runAllTimers();
-            expect(Widget.option.mock.calls.length).toBe(0);
-        });
+    it("ignores 3rd-party changes in nested default props", () => {
+        const component = shallow(
+            <ControlledComponent>
+                <NestedComponent defaultC="default" />
+            </ControlledComponent>
 
-        it("ignores 3rd-party changes in nested default props if parent object changes", () => {
-            mount(
-                <ControlledComponent>
-                    <NestedComponent defaultC="default" />
-                </ControlledComponent>
-            );
+        );
+        const nested = component.find(NestedComponent).dive();
```

Comment:
I prefer using `mount` in such cases

Reply:
In these tests `.shallow()` renders the widget component and `.dive()` renders the nested option component - that's enough (all the components are rendered). 

Moreover, using `.mount()` makes impossible to set options for non-root components:
```console
ReactWrapper::setProps() can only be called on the root
```
See https://github.com/airbnb/enzyme/issues/1289.